cornedbeef/src/main/java/com/swiftkey/cornedbeef/CoachMark.java

```diff
@@ -321,6 +345,7 @@ public Point getPos() {
         protected View tokenView;
         protected OnShowListener showListener;
         protected OnTimeoutListener timeoutListener;
+        protected boolean shouldDismissOnAnchorDetach;
```

Comment:
Should it be true by default? In which cases should it be false?

Reply:
by default we maintain the current behaviour; if we want coach marks to auto-dismiss we would call `setDismissOnAnchorDetach(true)` when building

Reply:
But the current behaviour makes it easy to introduce leaks, shouldn't we fix the problem by default and allow you to explicitly set a different behaviour if you know what you're doing?

Reply:
Normally, I would argue that we should keep the current behaviour as default since it makes migration easier. However, since this is going to a major version bump anyway (because of the API change) I think it would be OK to change the default.